src/features/session_replay/aggregate/index.js

```diff
@@ -223,8 +223,27 @@ export class Aggregate extends AggregateBase {
   getHarvestContents () {
     const agentRuntime = getRuntime(this.agentIdentifier)
     const info = getInfo(this.agentIdentifier)
-    const firstTimestamp = this.timestamp.event.first || this.timestamp.cycle.first
-    const lastTimestamp = this.timestamp.event.last || this.timestamp.cycle.last
+
+    // do not let the last node be a meta node, since this NEEDS to precede a snapshot
+    // we will manually inject it later if we find a payload that is missing a meta node
+    const payloadEndsWithMeta = this.events[this.events.length - 1]?.type === 4
+    if (payloadEndsWithMeta) {
+      this.hasMeta = false
+      this.events = this.events.slice(0, this.events.length - 1)
+    }
+
+    // do not let the first node be a snapshot node, since this NEEDS to be preceded by a meta node
+    // we will manually inject it if this happens
+    const payloadStartsWithSnapshot = this.events[0]?.type === 4
```

Comment:
These node types should be moved to a constant that is more descriptive so we can just look and tell what a node type `4` is, IMHO.